src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs

```diff
@@ -570,22 +579,31 @@ static IEnumerable<Instruction> DigProperties(IEnumerable<(PropertyDefinition pr
 
 				if (indexArg != null)
 				{
-					var indexType = property.GetMethod.Parameters[0].ParameterType.ResolveGenericParameters(propDeclTypeRef);
-					if (TypeRefComparer.Default.Equals(indexType, module.TypeSystem.String))
-						yield return Create(Ldstr, indexArg);
-					else if (TypeRefComparer.Default.Equals(indexType, module.TypeSystem.Int32) && int.TryParse(indexArg, out int index))
-						yield return Create(Ldc_I4, index);
+					if (property == null && int.TryParse(indexArg, out int index)) //array
+						yield return Create(Ldc_I4, index);							
 					else
-						throw new BuildException(BindingIndexerParse, lineInfo, null, indexArg, property.Name);
+					{
+						var indexType = property.GetMethod.Parameters[0].ParameterType.ResolveGenericParameters(propDeclTypeRef);
```

Comment:
This would be terrible to do in practice... But can you make C# indexers with multiple parameters?
```csharp
public string this[string key1, string key2]
{
    get => someTupleDictionary[(key1, key2)]; 
    set => someTupleDictionary[(key1, key2)] = value;
}
```

Reply:
well, this PR is about arrays, not indexers, but yeah, it doesn't cover multidimensional arrays 🙉🙉🙉